editor/src/clj/editor/code/view.clj

```diff
@@ -2325,15 +2325,20 @@
 
 (defn handle-scroll! [view-node ^ScrollEvent event]
   (.consume event)
-  (when (set-properties! view-node :navigation
-                         (data/scroll (get-property view-node :lines)
-                                      (get-property view-node :scroll-x)
-                                      (get-property view-node :scroll-y)
-                                      (get-property view-node :layout)
-                                      (get-property view-node :gesture-start)
-                                      (.getDeltaX event)
-                                      (.getDeltaY event)))
-    (hide-suggestions! view-node)))
+  (when (if (or (.isControlDown event) 
+                (.isMetaDown event))
+          (-> (ui/contexts (ui/main-scene))
+              (ui/invoke-handler (cond (pos? (.getDeltaY event)) :zoom-in
```

Comment:
Is there a better way to dispatch the event?

Reply:
If your action is executed in the context of some JavaFX Node (in this case, the node should be accessible from the event, or using `(g/node-value view-node :canvas)`), you should use this node instead of `(ui/main-scene)` for invoking a command. You probably should use `ui/run-command` here instead.